sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -1045,6 +1045,36 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
       Seq(Row(null), Row(null))
     )
 
+    checkAnswer(
+      df.selectExpr("array_position(array(1), 1.23D)"),
```

Comment:
same problem here. The test doesn't read any column from `df`, so we should use `OneRowRelation`.